compiler-rt/lib/tysan/tysan.cpp

```diff
@@ -131,6 +131,14 @@ static bool isAliasingLegalUp(tysan_type_descriptor *TDA,
           break;
       }
 
+      // This offset can't be negative. Therefore we must be accessing something
+      // partially inside the last type
+      // We shouldn't check this if we are on the first member, Idx will
+      // underflow The first member can be offset in rare cases such as
```

Comment:
```suggestion
      // underflow. The first member can be offset in rare cases such as
```

Reply:
Is it `Idx` that underflows or `OffsetA`? `Idx` will only get incremented AFAICT. Would this be cleared if combined with the search loop above, maybe with a flag indicating that we took the exit due to the offset of the member begin larger than OffsetA?